### PR TITLE
Update setup.md and troubleshooting.md

### PR DESCRIPTION
**Pull request by Ace-Bansal**
screenshot from 2019-01-05 00-42-13
Users have been facing this problem while setting up the dev environment. I've updated the setup docs to include the command `yarn --ignore-engines` which solves this problem.
I've also mentioned using node v10 or lower to avoid any problems as a pre-requisite to running the project.

### CODE REVIEW

**Comment by Ace-Bansal** [comment]
@ragesoss could you please review and point out any mistakes that I may have made. Thanks!

File: docs/setup.md [hunk 1]
```diff
@@ -38,6 +39,8 @@ You can also contact us on slack for any further queries.
 - If you face the error that `Sorry! Your operating is not supported by this script`
   - You can try running the system dependent scripts from setup directory, according to your system
   - You can try manual installation
+- After running 'gulp', if you face the error `Failed to load external module @babel/register`
```
Comment (ragesoss) [inline docs/setup.md]: Instead of putting `--ignore-engines` in the documentation here, the setup script itself should be changed to use that flag.

Also, the manual setup instructions below should be updated to include that flag.

File: package.json [hunk 2]
```diff
@@ -34,7 +34,7 @@
     "eslint-plugin-import": "^2.14.0",
     "eslint-plugin-jsx-a11y": "^6.1.2",
     "eslint-plugin-react": "^7.11.1",
-    "gulp": "^3.8.7",
+    "gulp": "^3.9.1",
```
Comment (ragesoss) [inline package.json]: The changes to package.json and yarn.lock should not be included in this PR.

File: package-lock.json [hunk 3]
```diff
@@ -0,0 +1,2011 @@
+{
```
Comment (ragesoss) [inline package-lock.json]: This file should not be added. We use yarn.lock instead.

**Comment by psinghal20** [comment]
@Ace-Bansal yes, that is correct. You can iterate that over all the shell scripts.

**Comment by Ace-Bansal** [comment]
Changes made in latest commit:
1. Include the --ignore-engines flag in all setup scripts.
2. Include the tag in the manual setup instructions in setup.md
3. Remove package-lock.json
4. Undo the changes done in package.json & yarn.lock by previous commit.

**Comment by psinghal20** [comment]
Looks good to me! Sage can have a look tomorrow and merge it.
